tools/topology/topology2/development/cavs-nocodec-asrc.conf

```diff
@@ -0,0 +1,694 @@
+			stream_name "SSP0 Capture"
+			pcm_id		4
+		}
+	}
+
+	{
+		index 	8
+		Object.Widget.host-copier.1 {
+			stream_name "SSP1 Capture"
+			pcm_id		$SSP1_PCM_ID
+			<include/components/src_format_s32_to_sxx_convert.conf>
+		}
+	}
+
+	{
+		index 	12
+		Object.Widget.host-copier.1 {
+			stream_name "SSP2 Capture"
+			pcm_id		5
+			<include/components/src_format_s32_to_sxx_convert.conf>
+		}
+	}
+]
+Object.Base.route [
+	{
+		source $DMIC0_DAI_COPIER
+		sink "asrc.1.1"
+	}
+
+	{
+		source "asrc.1.1"
+		sink	"host-copier.3.capture"
+	}
+
+	{
+		source	"dai-copier.SSP.NoCodec-0.capture"
+		sink	"host-copier.4.capture"
+	}
+
+	{
+		source	"dai-copier.SSP.NoCodec-1.capture"
+		sink	"asrc.7.1"
+	}
+
+	{
+		source	"asrc.7.1"
+		sink	"host-copier.$SSP1_PCM_ID.capture"
+	}
+
+	{
+		source	"dai-copier.SSP.NoCodec-2.capture"
+		sink	"asrc.11.1"
+	}
+
+	{
+		source	"asrc.11.1"
+		sink	"host-copier.5.capture"
+	}
+
+
+]
+
+Object.PCM.pcm [
+	{
+		name	"ASRC Capture 3"
+		id 3
+		direction	"capture"
+		Object.Base.fe_dai."ASRC Capture 3" {}
+
+		Object.PCM.pcm_caps."capture" {
+			name "ASRC Capture 3"
+			# only 32-bit capture supported now
+			formats 'S16_LE,S24_LE,S32_LE'
+			channels_min $NUM_DMICS
+			channels_max $NUM_DMICS
+			rate_min 8000
+			rate_max 192000
+		}
+	}
+
+	{
+		name	"Port 0 Capture"
+		id 4
+		direction	"capture"
+		Object.Base.fe_dai.1 {
+			name	"Port 0 Capture"
+		}
+		Object.PCM.pcm_caps."capture" {
+			name "SSP0 Capture"
+			# only 32-bit capture supported now
+			formats 'S16_LE,S24_LE,S32_LE'
+			rate_min 8000
+			rate_max 192000
+		}
+	}
+
+	{
+		name	"Port 2 Capture"
+		id 5
+		direction	"capture"
+		Object.Base.fe_dai.1 {
+			name	"Port 2 Capture"
+		}
+		Object.PCM.pcm_caps."capture" {
+			name "SSP2 Capture"
+			# only 32-bit capture supported now
+			formats 'S16_LE,S24_LE,S32_LE'
+			rate_min 8000
+			rate_max 192000
+		}
+	}
+]
+
+
+Object.PCM.pcm [
+	{
+		name	"$SSP0_PCM_NAME"
+		id $SSP0_PCM_ID
+		direction	"playback"
+		Object.Base.fe_dai.1 {
+			name	"$SSP0_PCM_NAME"
+		}
+
+		Object.PCM.pcm_caps.1 {
+			name "SSP0 Playback"
+			formats 'S16_LE,S24_LE,S32_LE'
+			rate_min 8000
+			rate_max 192000
+		}
+	}
+
+	{
+		name	"$SSP1_PCM_NAME"
+		id $SSP1_PCM_ID
+		direction	"duplex"
+		Object.Base.fe_dai.1 {
+			name	"$SSP1_PCM_NAME"
+		}
+
+		Object.PCM.pcm_caps.1 {
+			direction	"playback"
+			name "SSP1 Playback"
+			formats 'S16_LE,S24_LE,S32_LE'
+			rate_min 8000
+			rate_max 192000
+		}
+
+		Object.PCM.pcm_caps.2 {
+			direction	"capture"
+			name "SSP1 Capture"
+			formats 'S16_LE,S24_LE,S32_LE'
+			rate_min 8000
+			rate_max 192000
+		}
+	}
+
+	{
+		name	"$SSP2_PCM_NAME"
+		id $SSP2_PCM_ID
+		direction	"playback"
+		Object.Base.fe_dai.1 {
+			name	"$SSP2_PCM_NAME"
+		}
+
+		Object.PCM.pcm_caps.1 {
+			direction	"playback"
+			name "SSP2 Playback"
+			formats 'S16_LE,S24_LE,S32_LE'
+			rate_min 8000
+			rate_max 192000
+		}
+	}
+]
+
+
+
+# PCM0 --> ASRC--> SSP0
+
+# PCM1 --> ASRC--> SSP1
+
+# PCM2--> SSP2
```

Comment:
In my UPX SSP0 and SSP2 work, SSP1 doesn't, I have used used SSP0 and SSP2 in topologies for tgl  but I guess your test device is different.

Reply:
yes, I can get the right curve